File: mobile/lib/application/use_cases/camion/update_ubicacion_use_case.dart
import '../../../core/repositories/camion_repository.dart';

class UpdateUbicacionUseCase {
  UpdateUbicacionUseCase(this._repo);
  final CamionRepository _repo;

  Future<void> call(String camionId, double lat, double lng) =>
      _repo.updateUbicacion(camionId, lat, lng);
}

File: backend/src/main/java/com/ibarra/abastecida/application/usecase/UpdateUbicacionUseCase.java
package com.ibarra.abastecida.application.usecase;

import com.ibarra.abastecida.domain.port.CamionRepositoryPort;
import org.springframework.stereotype.Service;

import java.util.UUID;

/**
 * Use case: update a truck's location.
 */
@Service
public class UpdateUbicacionUseCase {

    private final CamionRepositoryPort camionRepository;

    public UpdateUbicacionUseCase(CamionRepositoryPort camionRepository) {
        this.camionRepository = camionRepository;
    }

    /**
     * @return true if the truck existed and was updated, false if not found
     */
    public boolean execute(UUID camionId, double lat, double lng) {
        if (camionRepository.findById(camionId).isEmpty()) {
            return false;
        }
        camionRepository.updateUbicacion(camionId, lat, lng);
        return true;
    }
}
